src/native/corehost/runtime_config.cpp

```diff
@@ -429,6 +429,46 @@ const pal::string_t& runtime_config_t::get_tfm() const
     return m_tfm;
 }
 
+const int runtime_config_t::get_compat_major_version_from_tfm() const
+{
+    assert(m_valid);
+
+    // TFM is in form
+    // - netcoreapp#.#  for <= 3.1
+    // - net#.#  for >= 5.0
+    // In theory it could contain a suffix like `net6.0-windows` (or more than one)
+    // or it may lack the minor version like `net6`. SDK will normalize this, but the runtime should not 100% rely on it
+
+    if (m_tfm.empty())
+        return runtime_config_t::unknown_version;
+
+    size_t majorVersionStartIndex;
+    if (m_tfm.rfind(_X("netcoreapp"), 0) == 0) // this is StartsWith("netcoreapp") in std speak
+    {
+        majorVersionStartIndex = 10;  // "netcoreapp".Length
```

Comment:
These lengths can all be computed at compile time. I'd update them as such. Even if it was: `STRLEN(_X("netcoreapp"))`, where `STRLEN` could be some compile time method. A simple template based solution could be:

```c++
template<size_t L>
inline constexpr size_t STRLEN(const pal::char_t (&)[L])
{
    return L - 1;
}
```